include/swift/AST/DiagnosticsSema.def

```diff
@@ -3012,6 +3012,11 @@ WARNING(optional_pattern_match_promotion,none,
         (Type, Type))
 WARNING(optional_to_any_coercion,none,
         "expression implicitly coerced from %0 to %1", (Type, Type))
+WARNING(iuo_to_any_coercion,none,
+        "coercion of implicitly unwrappable value of type %0 to %1 does not "
+        "unwrap optional", (Type, Type))
+NOTE(iuo_to_any_coercion_note,none,
+     "implicitly unwrapped %0 '%1' declared here", (DescriptiveDeclKind, StringRef))
```

Comment:
I think it makes more sense to add a separate note for function declarations which is going to say `%0 '%1' with implicitly unwrapped result type is declared here`. Also you should use `DeclName` or at least `Identifier` for both notes instead of a string.

Reply:
Hmm okay! I have added a new diagnostic note for functions. I tried using `Identifier`, however it would trigger an asset related to special names. `DeclName` makes sense, so I have switched to that instead of `StringRef`.

Reply:
Yes, it makes sense if the name was special (e.g. constructor or subscript) and you tried to do `getBaseIdentifier` but it should work for `getBaseName()` (at least after looking at `<<` operator implementation for `DeclBaseName` which uses `userFacingName()` inside).